### ISSUE TITLE
authentication error when loading a benchmark

### ISSUE THREAD

**Issue by fpour** [polaris-hub/polaris#112]
### Polaris version

dev (0.4.1)

### Python Version

3.11.7

### Operating System

Linux

### Installation

pip install polaris-lib  

### Description

I can successfully log in to the Polaris Hub via `PolarisHubClient().login()`. However, when I want to load a benchmark, I get an "Invalid authentication credentials" error.
In particular, I run `benchmark = po.load_benchmark("polaris/pkis2-ret-wt-c-1")`, and get the following error:
```
---------------------------------------------------------------------------
HTTPStatusError                           Traceback (most recent call last)
File ~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:142, in PolarisHubClient._base_request_to_hub(self, url, method, **kwargs)
    [141](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:141) try:
--> [142](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:142)     response.raise_for_status()
    [144](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:144) except HTTPStatusError as error:
    [145](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:145)     # With an internal server error, we are not sure the custom error-handling code on the hub is reached.

File ~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/httpx/_models.py:759, in Response.raise_for_status(self)
    [758](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/httpx/_models.py:758) message = message.format(self, error_type=error_type)
--> [759](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/httpx/_models.py:759) raise HTTPStatusError(message, request=request, response=self)

HTTPStatusError: Client error '401 Unauthorized' for url 'https://polarishub.io/api/v1/benchmark/polaris/pkis2-ret-wt-c-1'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401

The above exception was the direct cause of the following exception:

PolarisHubError                           Traceback (most recent call last)
Cell In[6], [line 4](vscode-notebook-cell:?execution_count=6&line=4)
      [1](vscode-notebook-cell:?execution_count=6&line=1) import polaris as po
      [3](vscode-notebook-cell:?execution_count=6&line=3) # Load the benchmark from the Hub
----> [4](vscode-notebook-cell:?execution_count=6&line=4) benchmark = po.load_benchmark("polaris/pkis2-ret-wt-c-1")
      [6](vscode-notebook-cell:?execution_count=6&line=6) # Get the train and test data-loaders
      [7](vscode-notebook-cell:?execution_count=6&line=7) train, test = benchmark.get_train_test_split()

File ~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/loader/load.py:69, in load_benchmark(path, verify_checksum)
     [66](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/loader/load.py:66) if not is_file:
     [67](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/loader/load.py:67)     # Load from the Hub
     [68](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/loader/load.py:68)     client = PolarisHubClient()
---> [69](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/loader/load.py:69)     return client.get_benchmark(*path.split("/"), verify_checksum=verify_checksum)
     [71](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/loader/load.py:71) with fsspec.open(path, "r") as fd:
     [72](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/loader/load.py:72)     data = json.load(fd)

File ~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:433, in PolarisHubClient.get_benchmark(self, owner, name, verify_checksum)
    [419](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:419) def get_benchmark(
    [420](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:420)     self, owner: Union[str, HubOwner], name: str, verify_checksum: bool = True
    [421](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:421) ) -> BenchmarkSpecification:
    [422](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:422)     """Load a benchmark from the Polaris Hub.
    [423](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:423) 
    [424](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:424)     Args:
   (...)
    [430](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:430)         A `BenchmarkSpecification` instance, if it exists.
    [431](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:431)     """
--> [433](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:433)     response = self._base_request_to_hub(url=f"/benchmark/{owner}/{name}", method="GET")
    [435](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:435)     # TODO (cwognum): Currently, the benchmark endpoints do not return the owner info for the underlying dataset.
    [436](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:436)     # TODO (jstlaurent): Use the same owner for now, until the benchmark returns a better dataset entity
    [437](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:437)     response["dataset"] = self.get_dataset(
    [438](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:438)         owner, response["dataset"]["name"], verify_checksum=verify_checksum
    [439](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:439)     )

File ~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:153, in PolarisHubClient._base_request_to_hub(self, url, method, **kwargs)
    [151](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:151)     response = response.json()
    [152](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:152)     response = json.dumps(response, indent=2, sort_keys=True)
--> [153](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:153)     raise PolarisHubError(
    [154](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:154)         f"The request to the Polaris Hub failed. See the error message below for more details:\n{response}"
    [155](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:155)     ) from error
    [156](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:156) # Convert the response to json format if the response contains a 'text' body
    [157](https://vscode-remote+ssh-002dremote-002boracle-005finteractive.vscode-resource.vscode-cdn.net/home/farimah/projects/graphium-private/scripts/arxiv/~/miniconda3/envs/graphium-v2/lib/python3.11/site-packages/polaris/hub/client.py:157) try:

PolarisHubError: The request to the Polaris Hub failed. See the error message below for more details:
{
  "message": "Invalid authentication credentials"
}
```

Also, this is the output when I open the link indicated in `HTTPStatusError: Client error '401 Unauthorized' for url 'https://polarishub.io/api/v1/benchmark/polaris/pkis2-ret-wt-c-1'`:

<img width="403" alt="Screenshot 2024-06-06 at 14 51 32" src="https://github.com/polaris-hub/polaris/assets/12398091/3afea03b-1bca-4f58-99b1-dd88749d0da7">

### Steps to reproduce

```
import polaris as po
from polaris.hub.client import PolarisHubClient

# Load the benchmark from the Hub
benchmark = po.load_benchmark("polaris/pkis2-ret-wt-c-1")
```

**Comment by cwognum** [comment]
This may be because the changes in #106 have not been released. Let me release a new version!

**Comment by cwognum** [comment]
I just released [`v0.5.0`](https://github.com/polaris-hub/polaris/releases/tag/0.5.0). Could you try again with this version (only available on PyPi for now!)

**Comment by fpour** [comment]
That solved the issue. Thanks a lot!
